### PR TITLE
Update/input component sanitize number string

### PR DESCRIPTION
**Pull request by Jonalogy**
- Rename `PositiveIntegerText` to `DigitsOnly`
- Introduced `NonZeroLeadingDigits`

### CODE REVIEW

**Comment by pei-wang** [comment]
any reason include the type? is it possible put the logic out of Input to where it used? 
cc: @LingweiWang

File: src/components/EnumValues.ts [hunk 1]
```diff
@@ -54,11 +54,18 @@ export enum InputType {
    * Only positive 0-9 allowed. EG.: 0, 1
    * No `scientific-notation`
    * */
-  PositiveIntegerText = "positiveIntegerText",
+  DigitsOnly = "digitOnly",
   /*
-   * Only positive & negative integer/decimal allowed. EG.: -1.1, -1, 0, 1, 1.1
```
Comment (nataliechinlw) [inline src/components/EnumValues.ts]: I think the convention here was comment above code, so I think this comment was for `DecimalText`.

Comment (Jonalogy) [reply src/components/EnumValues.ts]: Yes you are right

**Comment by Jonalogy** [comment]
@pei-wang @LingweiWang 
It was a team decision. Checkout the discussion [here](https://github.com/gdsace/molb-web/pull/976#discussion_r282330514)

**Pull request merged**